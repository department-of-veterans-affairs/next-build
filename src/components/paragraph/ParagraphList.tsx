import { FormattedParagraph } from '@/lib/drupal/queries'
import { PARAGRAPH_RESOURCE_TYPES } from '@/lib/constants/resourceTypes'
import { Wysiwyg as FormattedWysiwyg } from '@/components/wysiwyg/formatted-type'
import {
  HeadingLevel,
  WithCurrentHeadingLevel,
} from '@/components/heading/formatted-type'
import { getLastHeadingLevelFromHtml } from '@/lib/utils/getLastHeadingLevelFromHtml'
import { Paragraph } from './template'

/**
 * This component renders a list of paragraphs with the appropriate heading levels based
 * on the content of the paragraphs. For instance, if a WYSIWYG paragraph contains a
 * heading or multiple headings in its content, paragraphs after it that accept a heading
 * level will take that into account and use the next appropriate heading level.
 *
 * When a paragraph's heading level is not being affected by a previous paragraph's
 * heading level, that paragraph's heading level will fall back to `initialHeadingLevel`.
 */
export const ParagraphList = ({
  paragraphs,
  currentHeadingLevel = 'h1',
}: {
  paragraphs: Array<(FormattedParagraph & WithCurrentHeadingLevel) | null>
} & WithCurrentHeadingLevel) => {
  const paragraphElements = []

  paragraphs.forEach((paragraph, index) => {
    if (!paragraph) return

    paragraphElements.push(
      <Paragraph
        {...paragraph}
        key={paragraph.id || index}
        currentHeadingLevel={currentHeadingLevel}
      />
    )

    // If this paragraph has a heading in it, that is now our current heading level to
    // be used in subsequent paragraphs.
    const foundHeadingLevel = getLastHeadingLevelFromParagraph(paragraph)
    if (foundHeadingLevel) {
      currentHeadingLevel = foundHeadingLevel
    }
  })

  return paragraphElements
}

/**
 * For use in conjunction with the `ParagraphsWithManagedHeadingLevels` component, this
 * function determines how to parse the paragraph's content to determine the last heading
 * level used inside the content and returns it.
 */
export function getLastHeadingLevelFromParagraph(
  paragraph: FormattedParagraph
): HeadingLevel | undefined {
  switch (paragraph.type) {
    case PARAGRAPH_RESOURCE_TYPES.WYSIWYG:
    case PARAGRAPH_RESOURCE_TYPES.RICH_TEXT_CHAR_LIMIT_1000:
      return getLastHeadingLevelFromHtml((paragraph as FormattedWysiwyg).html)
    // Actually, we don't want to drill down into multiple levels to find headings
    // case PARAGRAPH_RESOURCE_TYPES.QA_GROUP:
    // case PARAGRAPH_RESOURCE_TYPES.QA_SECTION:
    //   // Drill down into the QaGroup or QaSection to find the last heading level
    //   return (paragraph as QaSection | QaGroup).questions.reduce((lastLevel, question) => {
    //     const questionLevel = getLastHeadingLevelFromParagraph(question as FormattedParagraph )
    //     return questionLevel ?? lastLevel
    //   }, undefined as HeadingLevel | undefined)
    // case PARAGRAPH_RESOURCE_TYPES.QA:
    //   return (paragraph as QaParagraph).answers.reduce((lastLevel, answer) => {
    //     const answerLevel = getLastHeadingLevelFromParagraph(answer as FormattedParagraph )
    //     return answerLevel ?? lastLevel
    //   }, undefined as HeadingLevel | undefined)
  }
}
