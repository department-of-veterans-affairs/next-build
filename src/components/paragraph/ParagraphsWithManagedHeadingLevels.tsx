import { FormattedParagraph } from '@/lib/drupal/queries'
import { PARAGRAPH_RESOURCE_TYPES } from '@/lib/constants/resourceTypes'
import { Wysiwyg as FormattedWysiwyg } from '@/components/wysiwyg/formatted-type'
import {
  HeadingLevel,
  WithHeadingLevel,
} from '@/components/heading/formatted-type'
import { getLastHeadingLevelFromHtml } from '@/lib/utils/getLastHeadingLevelFromHtml'
import { incrementHeadingLevel } from '@/components/heading/incrementHeadingLevel'
import { Paragraph } from './template'

/**
 * This component renders a list of paragraphs with the appropriate heading levels based
 * on the content of the paragraphs. For instance, if a WYSIWYG paragraph contains a
 * heading or multiple headings in its content, paragraphs after it that accept a heading
 * level will take that into account and use the next appropriate heading level.
 */
export const ParagraphsWithManagedHeadingLevels = ({
  paragraphs,
  initialHeadingLevel,
}: {
  paragraphs: Array<(FormattedParagraph & WithHeadingLevel) | null>
  initialHeadingLevel?: HeadingLevel
}) => {
  let currentHeadingLevel: HeadingLevel | undefined
  const paragraphElements = []

  paragraphs.forEach((paragraph, index) => {
    if (!paragraph) return

    paragraphElements.push(
      <Paragraph
        {...paragraph}
        key={paragraph.id || index}
        headingLevel={
          // If we have a heading level from a previous paragraph, use it to determine
          // the appropriate heading level for the next paragraph
          currentHeadingLevel
            ? incrementHeadingLevel(currentHeadingLevel)
            : initialHeadingLevel
        }
      />
    )

    const lastFoundHeadingLevel = getLastHeadingLevelFromParagraph(paragraph)
    if (lastFoundHeadingLevel) {
      currentHeadingLevel = lastFoundHeadingLevel
    }
  })

  return paragraphElements
}

/**
 * For use in conjunction with the `ParagraphsWithManagedHeadingLevels` component, this
 * function determines how to parse the paragraph's content to determine the last heading
 * level used inside the content and returns it.
 */
export function getLastHeadingLevelFromParagraph(
  paragraph: FormattedParagraph
): HeadingLevel | undefined {
  switch (paragraph.type) {
    case PARAGRAPH_RESOURCE_TYPES.WYSIWYG:
    case PARAGRAPH_RESOURCE_TYPES.RICH_TEXT_CHAR_LIMIT_1000:
      return getLastHeadingLevelFromHtml((paragraph as FormattedWysiwyg).html)
  }
}
