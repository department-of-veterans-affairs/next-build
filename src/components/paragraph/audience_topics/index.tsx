import { v4 as uuidv4 } from 'uuid'
import Link from 'next/link'

export const getTagsList = (fieldTags) => {
  if (!fieldTags) return null

  const {
    field_topics: fieldTopics = [],
    field_audience_beneficiares: fieldAudienceBeneficiares,
  } = fieldTags

  const topics = fieldTopics.map((topic) => ({
    ...topic,
    categoryLabel: 'Topics',
  }))

  const audiences = [fieldAudienceBeneficiares]
    .filter((tag) => !!tag)
    .map((audience) => ({
      ...audience,
      categoryLabel: 'Audience',
    }))

  const tagList = [...topics, ...audiences]

  return tagList.sort((a, b) => a.categoryLabel.localeCompare(b.categoryLabel))
}

const AudienceTopics = ({ paragraph }) => {
  if (!paragraph) paragraph = []
  // Build tag list
  const tags = getTagsList(paragraph)
  // only render tags where length is greater than 0
  if (tags.length === 0) return null

  const tagsList = tags.map((tag) => (
    <div key={uuidv4()}>
      <div className="vads-u-margin-right--1 vads-u-margin-bottom--1 medium-screen:vads-u-margin-bottom--0">
        <Link href={`${tag.path.alias}/${tag.name}`}>
          <a className="vads-u-margin-bottom--1p5 usa-button-secondary vads-u-font-size--sm vads-u-border--1px vads-u-border-color--primary vads-u-padding--0p25 vads-u-padding-x--0p5 vads-u-margin-left--1p5 vads-u-text-decoration--none vads-u-color--base">
            {tag.name}
          </a>
        </Link>
      </div>
    </div>
  )) // end map

  return (
    <div className="vads-u-padding-x--1 large-screen:vads-u-padding-x--0">
      <div className="vads-u-padding-top--3 vads-u-padding-bottom--1p5 vads-u-border-top--1px vads-u-border-bottom--1px vads-u-border-color--gray-light vads-u-display--flex vads-u-align-items--flex-start vads-u-flex-direction--row">
        <strong className="vads-u-margin-bottom--2 medium-screen:vads-u-margin-bottom--0">
          Tags:
        </strong>
        <div className="vads-u-display--flex vads-u-flex-wrap--wrap">
          {tagsList}
        </div>
      </div>
    </div>
  )
}

export default AudienceTopics
