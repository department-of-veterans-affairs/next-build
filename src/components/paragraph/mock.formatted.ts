import { FormattedParagraph } from '@/lib/drupal/queries'

export const nestedQaParagraphs: FormattedParagraph[] = [
  {
    type: 'paragraph--q_a_section',
    id: '9f11f1d1-47c9-4461-af9e-3b2155028e72',
    displayAccordion: false,
    header: null,
    intro: null,
    questions: [
      {
        type: 'paragraph--q_a',
        question: 'What is mpox?',
        answers: [
          {
            type: 'paragraph--wysiwyg',
            id: '21c7ab2b-6caa-438f-8133-f308bb37755b',
            html: '<p>Mpox is a rare disease caused by infection with the mpox virus. The mpox virus is part of the same family of viruses as the variola virus, the virus that causes smallpox. Mpox symptoms are similar to smallpox symptoms, but milder, and mpox is rarely fatal. Mpox is not related to chickenpox.</p>',
          },
        ],
        id: '56597b36-3c81-4ca3-92b2-465bc4121d34',
      },
      {
        type: 'paragraph--q_a',
        question: 'What are signs and symptoms of mpox?',
        answers: [
          {
            type: 'paragraph--wysiwyg',
            id: 'c12156c6-5dda-40fa-b2d2-b00e7254d253',
            html: '<p>Sign and symptoms</p>',
          },
          {
            type: 'paragraph--collapsible_panel',
            id: '68824e7d-b5cd-4f1e-a8b3-8af70c2bd51e',
            entityId: 114599,
            paragraphs: [
              {
                type: 'paragraph--collapsible_panel_item',
                id: '084bb12d-1f76-495e-8acc-4d1e805641c5',
                entityId: 114598,
                title: 'Symptoms of mpox',
                wysiwyg:
                  '<ul>\n' +
                  '\t<li>Fever</li>\n' +
                  '\t<li>Headache</li>\n' +
                  '\t<li>Muscle aches and backache</li>\n' +
                  '\t<li>Swollen lymph nodes</li>\n' +
                  '\t<li>Chills</li>\n' +
                  '\t<li>Exhaustion</li>\n' +
                  '\t<li>Respiratory symptoms (e.g. sore throat, nasal congestion, or cough)</li>\n' +
                  '\t<li>A rash that may be located on or near the genitals (penis, testicles, labia, and vagina) or anus (butthole) but could also be on other areas like the hands, feet, chest, face, or mouth.\n' +
                  '\t<ul>\n' +
                  '\t\t<li>The rash will go through several stages, including scabs, before healing.</li>\n' +
                  '\t\t<li>The rash can look like pimples or blisters and may be painful or itchy.</li>\n' +
                  '\t</ul>\n' +
                  '\t</li>\n' +
                  '</ul>\n',
                paragraphs: [],
              },
            ],
            bordered: false,
          },
        ],
        id: 'c48857ed-0660-4d22-82d3-75bac5b47495',
      },
      {
        type: 'paragraph--q_a',
        question: 'How does it spread?',
        answers: [
          {
            type: 'paragraph--wysiwyg',
            id: '32f3d59b-6b93-4d85-807c-748915624da5',
            html: '<p>Mpox spreads in a few ways.</p>',
          },
          {
            type: 'paragraph--collapsible_panel',
            id: '7f4bede0-9260-46ce-8a18-2eeb6ee2d3d6',
            entityId: 114605,
            paragraphs: [
              {
                type: 'paragraph--collapsible_panel_item',
                id: 'c7e743d5-fffe-450f-9983-88655ff3567f',
                entityId: 114602,
                title: 'Through close, personal, often skin-to-skin contact',
                wysiwyg:
                  '<ul><li>Mpox can spread to anyone through close, personal, often skin-to-skin contact, including:<ul><li>Direct contact with mpox rash, scabs, or body fluids from a person with mpox.</li><li>Touching objects, fabrics (clothing, bedding, or towels), and surfaces that have been used by someone with mpox.</li><li>Contact with respiratory secretions.</li></ul></li></ul>',
                paragraphs: [],
              },
              {
                type: 'paragraph--collapsible_panel_item',
                id: '40b8e8c5-d7dd-4658-bde8-d8e9f0c8d11a',
                entityId: 114603,
                title: 'Direct contact ',
                wysiwyg:
                  '<ul><li>This direct contact can happen during intimate contact, including:<ul><li>Oral, anal, and vaginal sex&nbsp;or touching the genitals (penis, testicles, labia, and vagina) or anus (butthole) of a person with mpox.</li><li>Hugging, massage, and kissing.</li><li>Prolonged face-to-face contact.</li><li>Touching fabrics and objects during sex that were used by a person with mpox and that have not been disinfected, such as bedding, towels, fetish gear, and sex toys.</li></ul></li></ul>',
                paragraphs: [],
              },
              {
                type: 'paragraph--collapsible_panel_item',
                id: '0c0ac5da-7535-4a91-93c9-adcd1da53a10',
                entityId: 114604,
                title: 'Pregnancy',
                wysiwyg:
                  '<ul><li>A pregnant person can spread the virus to their fetus through the placenta.</li></ul>',
                paragraphs: [],
              },
            ],
            bordered: false,
          },
          {
            type: 'paragraph--wysiwyg',
            id: 'd34372d4-3443-49bd-9847-e068df1a1bda',
            html: '<p>Itâ€™s also possible for people to get mpox from infected animals, either by being scratched or bitten by the animal or by preparing or eating meat or using products from an infected animal. A person with mpox can spread it to others from the time symptoms start until the rash has fully healed and a fresh layer of skin has formed. The illness typically lasts 2-4 weeks.</p><p>Scientists are still researching:</p><ul><li>If the virus can be spread when someone has no&nbsp;<a href="https://www.cdc.gov/mpox/signs-symptoms/index.html">symptoms</a></li><li>How often mpox is spread through respiratory secretions, or when a person with mpox symptoms might be more likely to spread the virus through respiratory secretions.</li><li>Whether mpox can be spread through semen, vaginal fluids, urine, or feces.</li></ul>',
          },
        ],
        id: '55a5cad7-f029-4765-9c98-88b27d84f2c8',
      },
      {
        type: 'paragraph--q_a',
        question: 'How can I protect myself and others?',
        answers: [
          {
            type: 'paragraph--wysiwyg',
            id: '41b015f6-1729-4a88-bd6c-43db22188da4',
            html: '<ul><li>Avoid close, skin-to-skin contact with people who have a rash that looks like mpox.<ul><li>Do not touch the rash or scabs of a person with mpox.</li><li>Do not kiss, hug, cuddle or have sex with someone with mpox.</li></ul></li><li>Avoid contact with objects and materials that a person with mpox has used.<ul><li>Do not share eating utensils or cups with a person with mpox.</li><li>Do not handle or touch the bedding, towels, or clothing of a person with mpox.</li></ul></li><li>Wash your hands often with soap and water or use an alcohol-based hand sanitizer, especially before eating or touching your face and after you use the bathroom.</li></ul><p>In Central and West Africa, avoid contact with animals that&nbsp;<a href="https://www.cdc.gov/mpox/causes/index.html">can spread mpox</a>&nbsp;virus, usually rodents and primates. Also, avoid sick or dead animals, as well as bedding or other materials they have touched.</p>',
          },
        ],
        id: 'f100ad55-90a5-415a-947f-ff97f074ac64',
      },
    ],
  },
]
