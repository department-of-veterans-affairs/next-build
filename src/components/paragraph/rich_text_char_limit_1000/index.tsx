import {
  ParagraphProps,
  ParagraphResourceType,
  ParagraphMetaInfo,
} from '@/types/paragraph'

import { isValidData } from '@/utils/helpers'
import Wysiwyg from '@/components/wysiwyg'

function RichTextCharLimit1000({ paragraph, className }: ParagraphProps) {
  if (!isValidData(paragraph)) return

  return (
    <div data-entity-id={paragraph.id}>
      <Wysiwyg key={paragraph.id} className={className} paragraph={paragraph} />
    </div>
  )
}

export const Meta: ParagraphMetaInfo = {
  resource: ParagraphResourceType.RichTextCharLimit1000,
  component: RichTextCharLimit1000,
}

export default RichTextCharLimit1000
