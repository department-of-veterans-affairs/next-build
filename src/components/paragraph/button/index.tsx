import Link from 'next/link'
import { ParagraphButton } from '@/types/paragraph'

function isRequestValid(paragraph: ParagraphButton) {
  return paragraph?.field_button_label && paragraph?.field_button_link?.uri
}

const Button = ({ paragraph }): JSX.Element => {
  if (!paragraph || !isRequestValid(paragraph)) return

  return (
    <div key={paragraph.id}>
      <Link href={paragraph.field_button_link?.uri} passHref>
        <a className="vads-c-action-link--blue">
          {paragraph.field_button_label}
        </a>
      </Link>
    </div>
  )
}

export default Button
