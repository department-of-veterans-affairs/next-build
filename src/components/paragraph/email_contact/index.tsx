import Link from 'next/link'
import { recordEvent } from '@/utils/recordEvent'
import { ParagraphEmailContact } from '@/types/paragraph'

function isRequestValid(paragraph: ParagraphEmailContact) {
  return (
    paragraph.field_email_label !== null &&
    paragraph.field_email_address !== null
  )
}

const EmailContact = ({ paragraph }): JSX.Element => {
  if (!paragraph || !isRequestValid(paragraph)) return

  const EMAIL_CONTACT_EVENT = {
    event: 'nav-linkslist',
    'links-list-header': `${encodeURIComponent(paragraph.field_email_address)}`,
    'links-list-section-header': 'Need more help?',
  }

  return (
    <div key={paragraph.id}>
      <strong>{paragraph.field_email_label}: </strong>
      <Link href={'mailto:' + paragraph.field_email_address} passHref>
        <a
          onClick={() => recordEvent(EMAIL_CONTACT_EVENT)}
          rel="noreferrer noopener"
        >
          {paragraph.field_email_address}
        </a>
      </Link>
    </div>
  )
}

export default EmailContact
