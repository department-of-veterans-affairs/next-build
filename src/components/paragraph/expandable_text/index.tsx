import { ParagraphExpandableText } from '@/types/paragraph'

function isRequestValid(paragraph: ParagraphExpandableText) {
  return paragraph.field_text_expander !== null
}

const ExpandableText = ({ paragraph }): JSX.Element => {
  if (!paragraph || !isRequestValid(paragraph)) return

  return (
    <div
      key={paragraph.id}
      className="form-expanding-group borderless-alert additional-info-container"
    >
      <div className="additional-info-title">
        {paragraph.field_text_expander}
      </div>

      {paragraph.field_wysiwyg && (
        <div className="additional-info-content usa-alert-text" hidden>
          {paragraph.field_wysiwyg.processed}
        </div>
      )}
    </div>
  )
}

export default ExpandableText
