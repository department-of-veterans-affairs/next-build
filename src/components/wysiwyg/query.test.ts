/**
 * @jest-environment node
 */

import { queries } from '@/lib/drupal/queries'
import { ParagraphWysiwyg } from '@/types/drupal/paragraph'
import { ParagraphRichTextCharLimit1000 } from '@/types/drupal/paragraph'
import wysiwygMockData from './mock.json'
import richTextMockData from './mock.richTextCharLimit1000.json'

const wysiwygMock = wysiwygMockData as ParagraphWysiwyg[]
const richTextMock = richTextMockData as ParagraphRichTextCharLimit1000[]

describe('paragraph--wysiwyg formatData', () => {
  test('outputs formatted data', () => {
    expect(
      wysiwygMock.map((mock) => {
        return queries.formatData('paragraph--wysiwyg', mock)
      })
    ).toMatchSnapshot()
  })
})

describe('paragraph--rich_text_char_limit_1000 formatData', () => {
  test('outputs formatted data', () => {
    expect(
      richTextMock.map((mock) => {
        return queries.formatData('paragraph--rich_text_char_limit_1000', mock)
      })
    ).toMatchSnapshot()
  })
})
