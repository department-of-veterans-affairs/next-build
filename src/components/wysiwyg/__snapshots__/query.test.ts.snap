// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`paragraph--rich_text_char_limit_1000 formatData outputs formatted data 1`] = `
[
  {
    "html": "<p>You may be eligible for VA disability benefits if you meet both of the requirements listed below.</p>
<p><strong>Both of these must be true. You: </strong></p>
<ul>
<li>Have an illness we believe is caused by contact with Agent Orange (called a presumptive disease), <strong>and</strong></li>
<li>Served in a location or job that put you in contact with Agent Orange (called having a presumption of contact) </li>
</ul>
<p>Read the full eligibility requirements for presumptive diseases and presumption of contact below.</p>
",
    "id": "c4d10aa1-3bb9-4f39-99fd-f110b465603f",
    "type": "paragraph--rich_text_char_limit_1000",
  },
  {
    "html": "<p>We believe Agent Orange can cause certain illnesses, including some types of cancer. We call these presumptive diseases. If you have one of these illnesses, you won’t need to show that the illness started during—or got worse because of—your military service. </p>
",
    "id": "78f74d21-4120-45be-9f07-73e03e6b8f54",
    "type": "paragraph--rich_text_char_limit_1000",
  },
]
`;

exports[`paragraph--wysiwyg formatData outputs formatted data 1`] = `
[
  {
    "html": "<p>We encourage you to visit when most convenient for you and the patient. There are no set visitor hours for most departments. Exceptions are listed below:</p>

<h4>Medical, surgical, and hospice wards</h4>

<p>24 hours a day, 7 days a week</p>

<h4>Behavioral health inpatient wards</h4>

<p>1:00 p.m. – 2:00 p.m. and 6:00 p.m. -7:30 p.m., 7 days a week</p>

<h4>Critical care</h4>

<p>Visiting hours will be individualized to meet the needs of the patient, family and caregivers.</p>

<h4>Post-anesthesia recovery</h4>

<p>Generally, visitors are not permitted. In certain circumstances, the doctor or nurse may make an exception.</p>

<h4>General visitor policies</h4>

<ul><li>Children under the age of 16 must be directly supervised by an adult, and bedside visits are limited to 15 minutes. </li>
	<li>Children under the ages of 16 are not permitted in the dialysis unit due to infection control and safety reasons. </li>
	<li>Children under the age of 16 are not permitted to visit the psychiatric units without the physician's permission.</li>
	<li>While in restraints, patients may not have visitors. </li>
	<li>Seriously ill and hospice patients may have visits unless the staff physician determines that the patient's condition is such that visits should be restricted to specific times.</li>
	<li>Staff may ask visitors to leave the room when performing patient care. </li>
	<li>Visitors are not permitted to bring weapons, cameras or other prohibited articles into the medical center except when conducting official business authorized by the Director or designee.</li>
	<li>Visitors with communicable diseases are not permitted in the medical center.</li>
	<li>Visitors must observe the infection control signs and should report to the nurses' station for instructions before entering a patient room.</li>
	<li>Visitors must observe hand hygiene policy guidelines requiring hand hygiene on entry and exit from patient rooms and at other times as directed by staff.</li>
	<li>Visitors may bring candy, fruit and other food items in limited quantities if it has been clarified by the nursing staff as appropriate for the patient's diet. </li>
	<li>Fresh fruit and flowers are not permitted in the Critical Care Center.</li>
	<li>Visitors are not permitted to eat or drink in the units.</li>
	<li>Members of the religious community are encouraged to visit patients who are members of their local churches and/or synagogues. However, under no circumstances are they permitted to do general visitation by going from bed to bed, ward to ward or unit to unit.</li>
</ul><p> </p>",
    "id": "58bd66a0-7f20-41e0-984a-975ee25a1aee",
    "type": "paragraph--wysiwyg",
  },
  {
    "html": "",
    "id": "a0e63a12-9a23-4e7f-95a0-b3a6ce45a9e9",
    "type": "paragraph--wysiwyg",
  },
]
`;
